### PR TITLE
[SPARK-48915][SQL][TESTS] Add some uncovered predicates(!=, <=, >, >=) in test cases of `GeneratedSubquerySuite`

### PR DESCRIPTION
**Pull request by wayneguow**
### What changes were proposed in this pull request?

This PR aims to add some predicates(!=, <=, >, >=) which are not covered in test cases of `GeneratedSubquerySuite`.

### Why are the changes needed?

Better coverage of current subquery tests in `GeneratedSubquerySuite`.
For more information about subqueries in `postgresq`, refer to:
https://www.postgresql.org/docs/current/functions-subquery.html#FUNCTIONS-SUBQUERY
https://www.postgresql.org/docs/current/functions-comparisons.html#ROW-WISE-COMPARISON

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass GA and Manual testing with `GeneratedSubquerySuite`.
image

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by wayneguow** [comment]
cc @cloud-fan @andylam-db

**Comment by wayneguow** [comment]
cc @cloud-fan @andylam-db

File: sql/core/src/test/scala/org/apache/spark/sql/QueryGeneratorHelper.scala [hunk 1]
```diff
@@ -96,7 +109,10 @@ trait QueryGeneratorHelper {
     // Subquery to be treated as a Relation.
     val RELATION = Value
     // Subquery is a Predicate - types of predicate subqueries.
-    val SCALAR_PREDICATE_EQUALS, SCALAR_PREDICATE_LESS_THAN, IN, NOT_IN, EXISTS, NOT_EXISTS = Value
+    val SCALAR_PREDICATE_EQUALS, SCALAR_PREDICATE_NOT_EQUALS,
+    SCALAR_PREDICATE_LESS_THAN, SCALAR_PREDICATE_LESS_THAN_OR_EQUALS,
+    SCALAR_PREDICATE_GREATER_THAN, SCALAR_PREDICATE_GREATER_THAN_OR_EQUALS,
+    IN, NOT_IN, EXISTS, NOT_EXISTS = Value
```
Comment (andylam-db) [inline sql/core/src/test/scala/org/apache/spark/sql/QueryGeneratorHelper.scala:115]: NIT: Indentation

**Review by andylam-db — COMMENTED** [review]
Thanks for the PR! 

Your implementation improves coverage of predicate types of scalar subquery in the WHERE clause, but I think the ticket is asking for different predicate types in the correlation predicate (within the subquery itself). Your PR is helpful of course, and I think we should merge it, but I think the ask on the ticket is different.

**Review by andylam-db — APPROVED** [review]

**Comment by wayneguow** [comment]
> Thanks for the PR!
> 
> Your implementation improves coverage of predicate types of scalar subquery in the WHERE clause, but I think the ticket is asking for different predicate types in the correlation predicate (within the subquery itself). Your PR is helpful of course, and I think we should merge it, but I think the ask on the ticket is different.

@andylam-db Thank you for your explanation. In addition to the current PR, I may be able to open a new PR.

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/querytest/GeneratedSubquerySuite.scala [hunk 2]
```diff
@@ -182,8 +182,11 @@ class GeneratedSubquerySuite extends DockerJDBCIntegrationSuite with QueryGenera
     } else {
```
Comment (wayneguow) [inline connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/querytest/GeneratedSubquerySuite.scala:182]: I should add some other predicates like `NotEquals`, `LessThan`... (reuse all newly added ones) here, is this right? @andylam-db

Comment (andylam-db) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/querytest/GeneratedSubquerySuite.scala:182]: Not sure I understand what you mean by `where the upper layer calls`.

Comment (wayneguow) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/querytest/GeneratedSubquerySuite.scala:182]: Oh, sorry, I didn't make it clear, I meant in line359-362, 
`for {
        subqueryOperator <- subqueryOperators
        isDistinct <- distinctChoices(subqueryOperator)
      }`

add another axis about `correlationConditions` here to cover all the different predicates, not only hardcoded about `=` in `generateQuery` method.

Comment (andylam-db) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/querytest/GeneratedSubquerySuite.scala:182]: Sounds great!

**Review by averyqi-db — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**